src/core/connection.c

```diff
@@ -703,6 +706,13 @@ QuicConnQueueOper(
         CXPLAT_DBG_ASSERT(QuicConnIsServer(Connection));
         CXPLAT_DBG_ASSERT(Connection->SourceCids.Next != NULL || CxPlatIsRandomMemoryFailureEnabled());
     }
+    if (Oper->Type == QUIC_OPER_TYPE_API_CALL) {
```

Comment:
Would it be better to have this assert when processing the operation rather than when queuing it?
Long term, if we apply this pattern, this function could become very messy, with validation for many different operations, instead of being completely operation agnostic.

Reply:
I did add another assert in the processing path (in the connection close function).

Reply:
In my experience, the more asserts the better. Different code paths hit differently.